Package.swift

```diff
@@ -1348,3 +1305,170 @@ func googleAppMeasurementDependency() -> Package.Dependency {
+        "CHANGELOG.md",
+        "CMakeLists.txt",
+        "Example/",
+        "LICENSE",
+        "Protos/CMakeLists.txt",
+        "Protos/Podfile",
+        "Protos/README.md",
+        "Protos/build_protos.py",
+        "Protos/cpp/",
+        "Protos/lib/",
+        "Protos/nanopb_cpp_generator.py",
+        "Protos/protos/",
+        "README.md",
+        "Source/CMakeLists.txt",
+        "Swift/",
+        "core/CMakeLists.txt",
+        "core/src/util/config_detected.h.in",
+        "core/test/",
+        "fuzzing/",
+        "test.sh",
+        // Swift PM doesn't recognize hpp files, so we're relying on search paths
+        // to find third_party/nlohmann_json/json.hpp.
+        "third_party/",
+
+        // Exclude alternate implementations for other platforms
+        "core/src/remote/connectivity_monitor_noop.cc",
+        "core/src/util/filesystem_win.cc",
+        "core/src/util/log_stdio.cc",
+        "core/src/util/secure_random_openssl.cc",
+      ],
+      sources: [
+        "Source/",
+        "Protos/nanopb/",
+        "core/include/",
+        "core/src",
+      ],
+      publicHeadersPath: "Source/Public",
+      cSettings: [
+        .headerSearchPath("../"),
+        .headerSearchPath("Source/Public/FirebaseFirestore"),
+        .headerSearchPath("Protos/nanopb"),
+        .define("PB_FIELD_32BIT", to: "1"),
+        .define("PB_NO_PACKED_STRUCTS", to: "1"),
+        .define("PB_ENABLE_MALLOC", to: "1"),
+        .define("FIRFirestore_VERSION", to: firebaseVersion),
+      ],
+      linkerSettings: [
+        .linkedFramework(
+          "SystemConfiguration",
+          .when(platforms: [.iOS, .macOS, .tvOS, .firebaseVisionOS])
+        ),
+        .linkedFramework("UIKit", .when(platforms: [.iOS, .tvOS, .firebaseVisionOS])),
+        .linkedLibrary("c++"),
+      ]
+    )
+  } else if ProcessInfo.processInfo.environment["FIREBASECI_USE_LOCAL_FIRESTORE_ZIP"] != nil {
+    // This is set when running `scripts/check_firestore_symbols.sh`.
+    return .binaryTarget(
+      name: "FirebaseFirestore",
+      // The `xcframework` should be moved to the root of the repo.
+      path: "FirebaseFirestore.xcframework"
+    )
+  }
+
+  return .binaryTarget(
+    name: "FirebaseFirestore",
+    url: "https://dl.google.com/firebase/ios/bin/firestore/10.11.0/FirebaseFirestore.zip",
+    checksum: "47da0c2d102c90d32ffa93cc5884c5a14eea006325944717b5e8d67ec9b440f3"
+  )
+}
+
+extension Platform {
+  // Xcode dependent value for the visionOS platform. Namespaced with
+  // "firebase" prefix to prevent any API collisions (such issues should not
+  // arise as the manifest APIs should be confined to the `Package.swift`).
+  static var firebaseVisionOS: Self {
+    #if swift(>=5.9)
+      // For Xcode 15, return the available `visionOS` platform.
+      return .visionOS
+    #else
+      // For Xcode 14, return `iOS` as visionOS is unavailable. Since all targets
+      // support iOS, this acts as a no-op.
+      return .iOS
```

Comment:
I assume Xcode 14 doesn't complain about `.iOS` being listed twice in the platforms array?

Reply:
Yep, manually tested that opening this branch's Package.swift in 14.3 resolved successfully.

There has been some precedent for featuring a platform twice in an array: https://github.com/google/GoogleDataTransport/blob/main/Package.swift#L109